membership-attribute-service/app/actions/AuthAndBackendViaIdapiAction.scala

```diff
@@ -0,0 +1,40 @@
+package actions
+
+import com.gu.identity.{IdapiService, SignedInRecently}
+import components.TouchpointBackends
+import filters.AddGuIdentityHeaders
+import play.api.mvc.{ActionRefiner, Request, Result, Results}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class AuthAndBackendViaIdapiAction(
+  touchpointBackends: TouchpointBackends,
+  howToHandleRecencyOfSignedIn: HowToHandleRecencyOfSignedIn
+)(
+  implicit ex: ExecutionContext
+) extends ActionRefiner[Request, AuthAndBackendRequest] {
+
+  override val executionContext = ex
+  override protected def refine[A](request: Request[A]): Future[Either[Result, AuthAndBackendRequest[A]]] =
+    touchpointBackends.normal.idapiService.RedirectAdvice.getRedirectAdvice(
+      request.headers.get(IdapiService.HeaderNameCookie).getOrElse(""),
+      request.headers.get(IdapiService.HeaderNameIdapiForwardedScope)
+    ).map(redirectAdvice => {
```

Comment:
Considering the pain of changing membership common, I'm not advocating for an immediate change, but I found the class name `RedirectAdviceResponse` slightly confusing. Isn't `HowToHandleRecencyOfSignedIn` using it to offer redirect advice,  rather than `RedirectAdviceResponse` offering the advice itself?

Reply:
`RedirectAdviceResponse` contains the actual _advice_ (the reason I'm using the word _advice_ is because the IDAPI endpoint isn't perfectly named, since it doesn't actually perform a redirect, just returns some JSON advising what redirect should take place)... hence the response is the advice and `HowToHandleRecencyOfSignedIn` is the instruction to `AuthAndBackendViaIdapiAction` in how to action that advice.

Reply:
Ok - just got a little confused as I felt that "how to handle recency of signed in" constituted part of that redirect advice.

Reply:
But see what you mean 👌